src/test/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilderUnitTest.java

```diff
@@ -406,4 +481,50 @@ public void testStdOut() {
         Assert.assertNotNull(((VCFWriter) writer).getOutputStream());
         Assert.assertNotEquals(((VCFWriter) writer).getStreamName(), IndexingVariantContextWriter.DEFAULT_READER_NAME);
     }
+
+  /**
+   * Create a named pipe, call "codeThatWritesToAFilename" with the name as argument,
+   * consumes the output as it's being run and returns the number of bytes that were output.
+   *
+   * This only works on Unix. Use this decoration for the tests that use it:
+   * @Test(singleThreaded = true, groups = "unix")
+   */
+    private long testWriteToPipe(final Consumer<String> codeThatWritesToAFilename) throws IOException, InterruptedException, ExecutionException {
+      // make the fifo
+      final File fifo = File.createTempFile("fifo.for.testWriteToPipe", "");
+      Assert.assertTrue(fifo.delete());
+      fifo.deleteOnExit();
+      final Process exec = new ProcessBuilder("mkfifo", fifo.getAbsolutePath()).start();
+      exec.waitFor(1, TimeUnit.MINUTES);
+      Assert.assertEquals(exec.exitValue(), 0, "mkfifo failed with exit code " + 0);
+
+      // run the code in a separate thread
+      ExecutorService executor = null;
+      try {
+        executor = Executors.newSingleThreadExecutor();
+        Future result = executor.submit(() -> {
+          try {
+            codeThatWritesToAFilename.accept(fifo.getAbsolutePath());
+          } catch (Exception x) {
+            // since the main thread is blocked reading the pipe, we may not notice the exception.
+            // Better to print it to the console to aid debugging.
+            x.printStackTrace();
+            throw(x);
+          }
+        });
+
+        // drain the pipe with the output.
+        InputStream inputStream = Files.newInputStream(fifo.toPath(), StandardOpenOption.READ);
+        int count = 0;
+        while (inputStream.read() >= 0) {
```

Comment:
If `codeThatWritesToAFilename` exits with an exception, this ends up blocking forever.  It's not ideal since that's the code that's actually under test and will probably start failing someday.  One simple but not fool proof thing to do would be to lift the reading into a background thread, and then wait on the writer to finish.  That will return immediately on a crash, but of course we'll have the same problem if the reader thread dies somehow.

Reply:
So maybe a `timeOut` to the `@Test` annotation for tests using them? Althought it can cause non-deterministic failures...

Reply:
Actually this won't block if `codeThatWritesToAFilename` exits with an exception. I wrote two test cases to demonstrate this.